### PR TITLE
updates to inventory plot dialog and R code

### PR DESCRIPTION
**Pull request by dannyparsons**

### CODE REVIEW

File: instat/static/InstatObject/R/data_object_R6.R [hunk 1]
```diff
@@ -1858,6 +1859,7 @@ data_object$set("public","split_date", function(col_name = "", week = FALSE, mon
     day_in_year_vector <- as.integer(yday(col_data))
 	  col_name <- next_default_item(prefix = "day_in_year", existing_names = self$get_column_names(), include_index = FALSE)
     self$add_columns_to_data(col_name = col_name, col_data = day_in_year_vector)
+    if(self$is_climatic_data()) self$set_climatic_types(types = c(doy = col_name))
```
Comment (dannyparsons) [inline instat/static/InstatObject/R/data_object_R6.R]: @stevekogo this auto adds climatic definition for doy - useful for inventory plot. Can you add for the other date variables in here? I just quickly did the ones needed for inventory.

**Comment by dannyparsons** [comment]
@stevekogo If you can update the dialog to the new code then I will merge this. But if it will take longer then I won't since the new code will make the current dialog unstable.

**Comment by stevekogo** [comment]
@dannyparsons, I am going to pull this branch then update the dialog then you will merge it in.

**Comment by dannyparsons** [comment]
fixes #2724

**Comment by dannyparsons** [comment]
A few design changes still needed on the dialog and tweaks to R code but needed to merge to avoid conflicts with other code.

**Pull request merged**